acpi/src/madt.rs

```diff
@@ -48,7 +50,55 @@ unsafe impl AcpiTable for Madt {
     }
 }
 
+pub type PhysToVirtFn = fn(u64) -> u64;
+
 impl Madt {
+    pub fn wakeup_aps(
```

Comment:
It feels odd to me to implement high-level functionality like booting APs directly on the `MADT`. This should probably be implemented in the `platform` layer - this should extract the physical address of the mailbox during its existing MADT parsing, and then provide this method on `PlatformInfo` I feel

Reply:
I wonder if it can be a separate function in the `platform` layer? It doesn't seem to belong to `PlatformInfo`.